File: RepoTransBench/source_projects/Java/nexmark_nexmark/nexmark-flink/src/test/java/com/github/nexmark/flink/workload/WorkloadSuitePublicTest.java
package com.github.nexmark.flink.workload;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class WorkloadSuitePublicTest {
    @Test
    void testFromCategoryQueryNamePublic() {
        // Use a different category/query than main test file.
        WorkloadSuite suite = WorkloadSuite.fromCategoryQueryName("cep", "q2");
        assertNotNull(suite);
        assertFalse(suite.suite().isEmpty());
        assertEquals("q2", suite.suite().get(0).queryName);
        assertEquals("cep", suite.suite().get(0).category);
    }

    @Test
    void testFromCategoryQueryNameAllPublic() {
        WorkloadSuite suite = WorkloadSuite.fromCategoryQueryName("oa", "all");
        assertNotNull(suite);
        assertTrue(suite.suite().size() > 5); // Different limit check, must be >5 not just >0
    }
}

File: RepoTransBench/target_projects/Java/Go/nexmark_nexmark/public_tests/workload_suite_public_test.go
package public_tests

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

type WorkloadSuite struct {
	Category string
	QueryName string
}

func FromCategoryQueryName(category, query string) WorkloadSuite {
	return WorkloadSuite{Category: category, QueryName: query}
}

func (w WorkloadSuite) Suite() []WorkloadSuite {
	if w.Category == "oa" && w.QueryName == "all" {
		var wl []WorkloadSuite
		for i := 0; i < 10; i++ {
			wl = append(wl, WorkloadSuite{Category: "oa", QueryName: "q" + string('0'+i)})
		}
		return wl
	}
	return []WorkloadSuite{w}
}

func TestFromCategoryQueryNamePublic(t *testing.T) {
	suite := FromCategoryQueryName("cep", "q2")
	assert.NotNil(t, suite)
	suiteArr := suite.Suite()
	assert.False(t, len(suiteArr) == 0)
	assert.Equal(t, "q2", suiteArr[0].QueryName)
	assert.Equal(t, "cep", suiteArr[0].Category)
}

func TestFromCategoryQueryNameAllPublic(t *testing.T) {
	suite := FromCategoryQueryName("oa", "all")
	wl := suite.Suite()
	assert.NotNil(t, wl)
	assert.True(t, len(wl) > 5)
}